caf/nrf_desktop: Improve DFU over SMP docs (#12742)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf52/developing.rst

Both the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` and :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT` Kconfig options automatically enable :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT`, which allows to build application update images for both slots.
To see which files are built when the option is enabled, go to the :ref:`app_build_mcuboot_output` page.

> .. fota_upgrades_nrfcdm_direct_xip_support_information_start

Reviewer: `fota_upgrades_over_ble_mcuboot_direct_xip_nrfcdm_note_start`,  `fota_upgrades_over_ble_mcuboot_direct_xip_nrfcdm_note_end` (to improve naming consistency)

To perform a FOTA update, complete the following steps:

.. fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start

1. Generate the DFU package by building your application with the FOTA support over Bluetooth Low Energy.

>    You can find the generated DFU archive in the following location :file:`<build_dir>/zephyr/dfu_application.zip`.

Reviewer 2: ```suggestion
   You can find the generated DFU archive in the following directory :file:`<build_dir>/zephyr/dfu_application.zip`.
```

Reviewer 3: Well, we passed the full path to the file. To be more precise, I suggest the following:

```suggestion
   You can find the generated :file:`dfu_application.zip` archive in the following directory :file:`<build_dir>/zephyr`.
```

How does it sound to you, @annwoj?

Reviewer 2: looks good!

   .. note::

>       Use a higher version for each image included in the DFU package as the FOTA process may be cancelled due to the downgrade prevention.

Reviewer 3: I guess I was too tired when composing :)

```suggestion
      For each image included in the DFU-generated package, you should use a higher version number than your currently active firmware, as your DFU target may reject the FOTA process due to a downgrade prevention mechanism.
```

Reviewer 3: @annwoj, please take a look also at my suggestion

Reviewer 2: Sounds good, though if we're going with this description, I'd split it into 2 sentence and do some minor adjustments, e.g.,:
For each image included in the DFU-generated package, use a higher version number than your currently active firmware. Otherwise, the DFU target may reject the FOTA process due to a downgrade prevention mechanism.

Reviewer 3: Sure, thanks for feedback

   |fota_upgrades_building|

> .. fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start

Author: First step in devguide says
 ```
Create a binary file that contains the new image. 
To create a binary file for an application update, build the application with the :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` option enabled.
```

We are already describing building the application, what files it produces and configuration before the dfu steps so we assume at this point that the application is already built.

> .. fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start

Reviewer 3: Hmmm, I see. I think we can update this step as it is too specific and doesn't fit.

```
1. Generate the DFU package by building your application with the FOTA support over Bluetooth Low Energy.

   You can find the generated DFU archive in the following location :file:`<build_dir>/zephyr/dfu_application.zip`.

.. note:
   Please use a higher version for each image included in the DFU package as the FOTA process may be cancelled due to the downgrade prevention.
```

> .. fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start

Reviewer 3: Would it also make it possible to integrate it fully within the nRF Desktop application?

> .. fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start

Author: Your suggestion looks good. I'll insert it, thanks.
That way all points will fit the nRF Desktop/CAF docs so there won't be a problem with rendering (i hope).

   |fota_upgrades_building|

> .. fota_upgrades_nrfcdm_common_dfu_steps_start

Reviewer: `fota_upgrades_over_ble_nrfcdm_common_dfu_steps_start`?
